### PR TITLE
gh-24140: Fixed some broken wikipedia links to anchor tags.

### LINKED ISSUE #24140: Fix broken links in the docs
If you run `make linkcheck` in the docs, there are several links that are broken and should be fixed. There's also a bunch of permanent redirects that can be updated.  

There are also several links to `docs.sympy.org`. These should be replaced with relative links using Sphinx (it would be nice if there were an automated check for this).

### PR DESCRIPTION
**Pull request by theWiseAman**
References to other Issues or PRs

Fixes #24140 

#### Brief description of what is fixed or changed

Fixed some broken wikipedia links to anchor tags.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by 1e9abhi1e10** [comment]
> References to other Issues or PRs
> 
> Fixes #24140
> 
> #### Brief description of what is fixed or changed
> #### Other comments
> #### Release Notes

Add a release note or If there is no release notes entry, write NO ENTRY.

**Comment by 1e9abhi1e10** [comment]
for the authors' test, you need to add your name and email in the `.mailmap` file and then run `python bin/mailmap_check.py` after adding your name and email. Read instructions [here ](https://github.com/sympy/sympy/wiki/Development-workflow#add-your-name-and-email-address-to-the-mailmap-file)

**Comment by 1e9abhi1e10** [comment]
> References to other Issues or PRs
> 
> Fixes #24140
> 
> #### Brief description of what is fixed or changed
> #### Other comments
> #### Release Notes

As said, add a release note else write `NO ENTRY`

File: .mailmap [hunk 1]
```diff
@@ -280,6 +280,7 @@ Alpesh Jamgade <alpeshjamgade21@gmail.com>
 Alsheh <alsheh@rpi.edu>
 Aman <amanmourya295@gmail.com>
 Aman Deep <amandeep1024@gmail.com>
+Aman Kumar Shukla <77196231+theWiseAman@users.noreply.github.com>
```
Comment (1e9abhi1e10) [inline .mailmap:283]: Add your personal email as follows:
`Joe Bloggs <joe@bloggs.com> joeb <1785690389+joeb@users.noreply.github.com>`

Comment (theWiseAman) [reply .mailmap]: Done

**Comment by theWiseAman** [comment]
@1e9abhi1e10 Can you please merge the changes until now? Other broken links are research paper links and I am having difficulty in finding suitable alternate links.

**Pull request merged**